src/strategyHelpers/canary/podCanaryHelper.ts

```diff
@@ -8,9 +8,13 @@ import * as canaryDeploymentHelper from './canaryHelper'
 import {isDeploymentEntity} from '../../types/kubernetesTypes'
 import {getReplicaCount} from '../../utilities/manifestUpdateUtils'
 
-export async function deployPodCanary(filePaths: string[], kubectl: Kubectl) {
+export async function deployPodCanary(
+   filePaths: string[],
+   kubectl: Kubectl,
+   onlyDeployStable: boolean = false
```

Comment:
In general boolean parameters to functions should just be a separate function.

deployPodCanary
deployPodCanaryStableOnly

more like this but this is an important fix (plus other things can be fixed up so we can just wait on a refactor)

Reply:
yeah there's stuff like it scattered all over the codebase, was gonna wait to just do a bulk fix + include testing and do that all at once